doc/sphinx/language/core/records.rst

```diff
@@ -136,13 +139,55 @@ synonym :cmd:`Structure`.
       .. coqtop:: all
 
          Record MyRecord := { myfield : nat } as VarName.
-         About myfield. (* observe the MyRecord variable is named VarName *)
+         About myfield. (* observe the MyRecord variable is named "VarName" *)
          
-         (* This allows us to make VarName implicit for instance, without having to rename the variable as would be the case without the as clause. *)
+         (* make "VarName" implicit without having to rename the variable,
+            which would be necessary without the "as" clause *)
          Arguments myfield {VarName}.   (* make "VarName" an implicit parameter *)
          Check myfield.
          Check (myfield (VarName:={| myfield := 0 |})).
 
+   .. exn:: Records declared with the keyword Record or Structure cannot be recursive.
+
+      The record name :token:`ident` appears in the type of its fields, but uses
+      the :cmd:`Record` command. Use  the :cmd:`Inductive` or
+      :cmd:`CoInductive` command instead.
+
+   .. exn:: Cannot handle mutually (co)inductive records.
+
+      Records cannot be defined as part of mutually inductive (or
+      coinductive) definitions, whether with records only or mixed with
+      standard definitions.
```

Comment:
It appears that this error message actually doesn't exist anymore because the first case ("with records only") is supported now. So the new error message to document instead is: "Mixed record-inductive definitions are not allowed".

Reply:
I dropped this message.  The "Mixed ..." message is not primarily for Record, which doesn't allow constructors, so let's address it later.

Reply:
Indeed, though it turns out that (currently) it can be raised by the `Record` command. But fine to keep this for the inductive chapter.

Reply:
Since it sounds like Hugo will relax this restriction, likely dropping the message entirely, I'll do nothing on this now.